src/server/game/Spells/Spell.cpp

```diff
@@ -5498,7 +5498,7 @@ SpellCastResult Spell::CheckCast(bool strict)
             }
             case SPELL_AURA_MOUNTED:
             {
-                if (m_caster->IsInWater())
+                if (m_caster->IsInWater() && m_spellInfo->HasAura(SPELL_AURA_MOD_INCREASE_MOUNTED_FLIGHT_SPEED))
```

Comment:
@Takenbacon : I remember running around on a DK character, exploring while riding on [[Path of Frost](http://www.wowhead.com/spell=3714/path-of-frost)] on the sea. When I for some reason found myself submerged in the water, I used to jump up on top of the Path of Frost again to mount. Not sure if that changed in one of the latest patches to WoTLK, but I remember it like you do.